codegen/templates/endpoint.tmpl

```diff
@@ -160,6 +156,27 @@ func (h *{{$handlerName}}) HandleRequest(
 	cliRespHeaders, err := w.Handle(ctx, req.Header, &requestBody)
 	{{else}}
 	response, cliRespHeaders, err := w.Handle(ctx, req.Header, &requestBody)
+
+	// log downstream response to endpoint
+	if ce := req.Logger.Check(zapcore.DebugLevel, "stub"); ce != nil {
+		zfields := []zapcore.Field{
+			zap.String("endpoint", h.endpoint.EndpointName),
+		}
+		{{- if ne .ResponseType ""}}
+		if response != nil {
```

Comment:
I'm going to drop this check

some response returned as primary type (like string, int), this will break when compare nil

directly marshal nil struct seems acceptable 
```
	bytes, err := json.Marshal(x)
    // bytes -> `null`
    // err -> `<nil>`
```